aspnetcore/fundamentals/minimal-apis/bindStreamPipeReader/7.0-samples/PipeStreamToBackgroundQueue/BackgroundQueueService.cs

```diff
@@ -0,0 +1,43 @@
+using System.Text.Json;
+using System.Threading.Channels;
+
+namespace BackgroundQueueService;
+
+class BackgroundQueue : BackgroundService
+{
+    private readonly Channel<Stream> _queue;
+    private readonly ILogger<BackgroundQueue> _logger;
+
+    public BackgroundQueue(Channel<Stream> queue, ILogger<BackgroundQueue> logger)
+    {
+        _queue = queue;
+        _logger = logger;
+    }
+    
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        await foreach (var dataStream in _queue.Reader.ReadAllAsync(stoppingToken))
+        {
+            // reset the stream to the beginning
+            dataStream.Position = 0;
```

Comment:
```suggestion
```

Can we assume this has been done. We should change this to a Memory<byte> though.

Reply:
alright.. let me try that out

Reply:
done

Reply:
It makes sense now on why it should be a memory<byte> and not a stream 😅

Reply:
> It makes sense now why it should be a memory<byte> and not a stream 😅